src/transformers/models/auto/modeling_auto.py

```diff
@@ -228,6 +228,13 @@
     ]
 )
 
+MODEL_FOR_IMAGE_SEGMENTATION_MAPPING_NAMES = OrderedDict(
+    [
+        # Model for Image Segmentation mapping
+        ("detr", "DetrForSegmentation"),
```

Comment:
`DetrForSegmentation` was probably not the best name. Could we add an alias called `DetrForImageSegmentation`? Or should we distinguish between the different kinds of segmentation (this particular one is panoptic segmentation)

Reply:
If I'm not mistaken, as noted in [this discussion](https://github.com/huggingface/transformers/pull/13828#discussion_r721652671), we will not make any distinction between different segmentation subtasks and expect models/architectures to implement post_process_segmentation method that implements "the most detailed version of subtask" they can.

More details:
This output shape
```py
List(Dict(
    mask: str, // base64 str
    label: float,
    score: int,
    parent: int // -1 for all for DetrForSegmentation
))
# removing `id` field from dict since it is no longer needed
```
will allow us to implement `part-aware panoptic` segmentation (which is the most detailed segmentation subtask). And if using this output, a user can construct `semantic`, `instance` or any othre segmentation subtask if they want